middleware/validate.js

```diff
@@ -0,0 +1,211 @@
+const knex = require('../db');
+knex('products')
+  .then(console.log);
+module.exports = {
+  validator: function (req, res, next) {
+
+    /************************
+     *  ARTICLES
+    ************************/
+    if (req.originalUrl.slice(0, 9) === '/articles') {
+      const encodedTitle = encodeURIComponent(req.params.title);
+      const title = decodeURIComponent(req.params.title);
+
+      // GET '/:title'
+      if (req.method === 'GET' && req.url === `/${encodedTitle}`) {
+        return knex('articles').where({ title: title })
+          .then((item) => {
+            if (item.length < 1) {
+              res.status(500);
+              return res.json({ "Error": "Couldn't find that article in our database." });
+            }
+            return next();
+          })
+          .catch((err) => {
+            console.log(err);
+            next();
```

Comment:
So instead of doing console.log with the err, you can actually pass the error into the `next()` function. This will propagate the error up to the server file where we can handle it there. Take a look at this doc for more info on that: 
https://expressjs.com/en/guide/error-handling.html